webapp/scss.md

```diff
@@ -0,0 +1,73 @@
+# SCSS - Code Guidelines
+
+## We use classes over tags and ids
+
+Never apply styles to `tags` or `ids` – use `classes` instead!
+
+Why?
+- HTML tags are responsible for the document _structure_, not the looks
+- targeting HTML tags comes with performance issues
+- ids are responsible for identifying a unique element, not for styling it
+- ids have higher specificity than classes and therefore don't play well together
+- classes can be combined and reused while ids are unique
+
+For more background see the following articles on [why not to style tags](https://frontstuff.io/you-need-to-stop-targeting-tags-in-css) and [why not to style ids](https://dev.to/clairecodes/reasons-not-to-use-ids-in-css-4ni4).
+
+## We use design tokens instead of magic numbers
+
+In order to achieve a consistent look and feel we use a set of pre-defined `design tokens` to style our components, for example `colors`, `sizes` and `box-shadows`. These tokens are stored as `SCSS variables` and reused throughout the app.
+
+So, instead of typing _pixel values_ or _hex codes_ make sure you use design tokens such as `height-header` or `color-input-border`.
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-31T10:45:00Z" title="Thursday, October 31st 2019, 11:45:00 am +01:00">Oct 31, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,73 @@
+# SCSS - Code Guidelines
+
+## We use classes over tags and ids
+
+Never apply styles to `tags` or `ids` – use `classes` instead!
+
+Why?
+- HTML tags are responsible for the document _structure_, not the looks
+- targeting HTML tags comes with performance issues
+- ids are responsible for identifying a unique element, not for styling it
+- ids have higher specificity than classes and therefore don't play well together
+- classes can be combined and reused while ids are unique
+
+For more background see the following articles on [why not to style tags](https://frontstuff.io/you-need-to-stop-targeting-tags-in-css) and [why not to style ids](https://dev.to/clairecodes/reasons-not-to-use-ids-in-css-4ni4).
+
+## We use design tokens instead of magic numbers
+
+In order to achieve a consistent look and feel we use a set of pre-defined `design tokens` to style our components, for example `colors`, `sizes` and `box-shadows`. These tokens are stored as `SCSS variables` and reused throughout the app.
+
+So, instead of typing _pixel values_ or _hex codes_ make sure you use design tokens such as `height-header` or `color-input-border`.
```


I had already started doing this after realizing we had design tokens, but as you pointed out in a `PR` review, I wasn't taking the opportunity to refactor other people's CSS when I touched it.

Reply:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2019-11-04T09:09:12Z" title="Monday, November 4th 2019, 10:09:12 am +01:00">Nov 4, 2019</time>_

---


Nobody's perfect. Sorry if I am too pedantic sometimes! :blush: